module/purpose/PS3Walk/src/PS3Walk.cpp

```diff
@@ -140,26 +235,18 @@ namespace module::purpose {
         });
 
         // output walk command based on updated strafe and rotation speed from joystick
-        // TODO(HardwareTeam): potential performance gain: ignore if value hasn't changed since last emit?
         on<Every<20, Per<std::chrono::seconds>>>().then([this] {
-            if (!headLocked) {
+            if (!head_locked) {
                 // Create a unit vector in the direction the head should be pointing
-                double yaw           = headYaw / std::numeric_limits<short>::max() * 1.5f;
-                double pitch         = headPitch / std::numeric_limits<short>::max();
-                Eigen::Vector3d uPCt = (Eigen::AngleAxisd(yaw, Eigen::Vector3d::UnitZ())
-                                        * Eigen::AngleAxisd(-pitch, Eigen::Vector3d::UnitY()))
+                Eigen::Vector3d uPCt = (Eigen::AngleAxisd(head_yaw, Eigen::Vector3d::UnitZ())
+                                        * Eigen::AngleAxisd(-head_pitch, Eigen::Vector3d::UnitY()))
                                            .toRotationMatrix()
                                        * Eigen::Vector3d::UnitX();
-                emit<Task>(std::make_unique<Look>(uPCt, false));
+                emit<Task>(std::make_unique<Look>(uPCt, true));
             }
 
-            // If moving, then determine the walk command and request it
             if (moving) {
-                const auto strafe_norm           = strafe / std::numeric_limits<short>::max();
-                const auto rotational_speed_norm = rotationalSpeed / std::numeric_limits<short>::max();
-                emit<Task>(
-                    std::make_unique<Walk>(Eigen::Vector3d(strafe_norm[0], strafe_norm[1], rotational_speed_norm)),
-                    1);
+                emit<Task>(std::make_unique<Walk>(walk_command.cast<double>()), 2);
```

Comment:
maybe put in a short comment saying that `walk_command` is changed in the joystick case above